CPScripts/watchdog.sh

```diff
@@ -33,6 +34,100 @@ pid=$(ps aux | grep "watchdog mariadb"  | grep -v grep | awk '{print $2}')
 		echo -e "\nWatchDog for MariaDB is running...\n"
 		echo $(ps aux | grep "watchdog mariadb"  | grep -v grep)
 	fi
+
+echo -e "\nChecking PowerDNS ..."
+pid=$(ps aux | grep "watchdog powerdns"  | grep -v grep | awk '{print $2}')
+	if [[ "$pid" == "" ]] ; then
+		echo -e "\nWatchDog for PowerDNS is gone , restarting..."
+		nohup watchdog powerdns > /dev/null 2>&1 &
+		echo -e "\nWatchDog for PowerDNS has been started..."
+	else
+		echo -e "\nWatchDog for PowerDNS is running...\n"
+		echo $(ps aux | grep "watchdog powerdns"  | grep -v grep)
+	fi
+
+echo -e "\nChecking Dovecot ..."
+pid=$(ps aux | grep "watchdog dovecot"  | grep -v grep | awk '{print $2}')
+	if [[ "$pid" == "" ]] ; then
+		echo -e "\nWatchDog for Dovecot is gone , restarting..."
+		nohup watchdog dovecot > /dev/null 2>&1 &
+		echo -e "\nWatchDog for Dovecot has been started..."
+	else
+		echo -e "\nWatchDog for Dovecot is running...\n"
+		echo $(ps aux | grep "watchdog dovecot"  | grep -v grep)
+	fi
+
+echo -e "\nChecking Postfix ..."
+pid=$(ps aux | grep "watchdog postfix"  | grep -v grep | awk '{print $2}')
+	if [[ "$pid" == "" ]] ; then
+		echo -e "\nWatchDog for Postfix is gone , restarting..."
+		nohup watchdog postfix > /dev/null 2>&1 &
+		echo -e "\nWatchDog for Postfix has been started..."
+	else
+		echo -e "\nWatchDog for Postfix is running...\n"
+		echo $(ps aux | grep "watchdog postfix"  | grep -v grep)
+	fi
+
+echo -e "\nChecking Pure-FTPd ..."
+pid=$(ps aux | grep "watchdog pureftpd"  | grep -v grep | awk '{print $2}')
+	if [[ "$pid" == "" ]] ; then
+		echo -e "\nWatchDog for Pure-FTPd is gone , restarting..."
+		nohup watchdog pureftpd > /dev/null 2>&1 &
+		echo -e "\nWatchDog for Pure-FTPd has been started..."
+	else
+		echo -e "\nWatchDog for Pure-FTPd is running...\n"
+		echo $(ps aux | grep "watchdog pureftpd"  | grep -v grep)
+	fi
+}
+
+check_service() {
```

Comment:
I moved all of this into its own function so that a service can have conditions for being checked.